modules/graphman-bundle.js

```diff
@@ -1,48 +1,76 @@
+/*
+ * Copyright ©  2024. Broadcom Inc. and/or its subsidiaries. All Rights Reserved.
+ */
 
 const utils = require("./graphman-utils");
-const SCHEMA_METADATA = require("./graphman").schemaMetadata();
-const GOID_PLURAL_METHODS = ["fipUsers", "federatedUsers", "internalUsers", "fipGroups", "federatedGroups", "internalGroups", "ldaps", "ldapIdps", "fips", "federatedIdps", "trustedCerts"];
-const DEPRECATED_TYPES = [
-    "webApiServices", "soapServices", "internalWebApiServices", "internalSoapServices",
-    "policyFragments", "globalPolicies", "backgroundTaskPolicies",
-    "fips", "ldaps", "fipUsers", "fipGroups"
-];
+const graphman = require("./graphman");
+const metadata = require("./graphman").schemaMetadata();
+const fileSuffixes = {
+    policies: '.policy',
+    services: '.service',
+    policyFragments: '.policy',
+    webApiServices: '.webapi',
+    soapServices: '.soap',
+    internalWebApiServices: '.internal-webapi',
+    internalSoapServices: '.internal-soap',
+    globalPolicies: '.global',
+    backgroundTaskPolicies: '.bgpolicy'
+};
 
 module.exports = {
     EXPORT_USE: 'export',
     IMPORT_USE: 'import',
-    GOID_MAPPING_PLURAL_METHODS: GOID_PLURAL_METHODS,
-    ENTITY_TYPE_PLURAL_TAG_FRIENDLY_NAME: {
-        policies: 'policy',
-        services: 'service',
-        policyFragments: 'policy',
-        webApiServices: 'webapi',
-        soapServices: 'soap',
-        internalWebApiServices: 'internal-webapi',
-        internalSoapServices: 'internal-soap',
-        globalPolicies: 'global',
-        backgroundTaskPolicies: 'bgpolicy'
+
+    /**
+     * Recommended way to iterate through the bundled entities.
+     * Callback will be invoked for every class of entities in the bundle. They will be invoked with the key, entities and type-info as arguments
+     * @param bundle input bundle
+     * @param knownEntitiesCallback callback function for the known entities
+     * @param unknownEntitiesCallback callback function for the unknown entities
+     */
+    forEach: function (bundle, knownEntitiesCallback, unknownEntitiesCallback) {
+        Object.entries(bundle).forEach(([key, entities]) => {
+            const typeInfo = graphman.typeInfoByPluralName(key);
+            if (typeInfo) {
+                if (entities.length) knownEntitiesCallback(key, entities, typeInfo);
```

Comment:
should we have null check for knownEntitiesCallback as well ?

Reply:
**knownEntitiesCallback** is mandatory.

Reply:
OK